webapp/channels/src/components/integrations/abstract_outgoing_webhook.tsx

```diff
@@ -171,67 +185,73 @@ export default class AbstractOutgoingWebhook extends React.PureComponent {
             team_id: this.props.team.id,
             channel_id: this.state.channelId,
             trigger_words: triggerWords,
-            trigger_when: parseInt(this.state.triggerWhen, 10),
+            trigger_when: this.state.triggerWhen,
             callback_urls: callbackUrls,
             display_name: this.state.displayName,
             content_type: this.state.contentType,
             description: this.state.description,
             username: this.state.username,
             icon_url: this.state.iconURL,
+            id: this.props.initialHook?.id || '',
+            create_at: this.props.initialHook?.create_at || 0,
+            update_at: this.props.initialHook?.update_at || 0,
+            delete_at: this.props.initialHook?.delete_at || 0,
+            creator_id: this.props.initialHook?.creator_id || '',
+            token: this.props.initialHook?.token || '',
         };
 
         this.props.action(hook).then(() => this.setState({saving: false}));
     };
 
-    updateDisplayName = (e) => {
+    updateDisplayName: ChangeEventHandler<HTMLInputElement> = (e) => {
         this.setState({
             displayName: e.target.value,
         });
     };
 
-    updateDescription = (e) => {
+    updateDescription: ChangeEventHandler<HTMLInputElement> = (e) => {
         this.setState({
             description: e.target.value,
         });
     };
 
-    updateContentType = (e) => {
+    updateContentType: ChangeEventHandler<HTMLSelectElement> = (e) => {
         this.setState({
             contentType: e.target.value,
         });
     };
 
-    updateChannelId = (e) => {
+    updateChannelId: ChangeEventHandler<HTMLSelectElement> = (e) => {
         this.setState({
             channelId: e.target.value,
         });
     };
 
-    updateTriggerWords = (e) => {
+    updateTriggerWords: ChangeEventHandler<HTMLTextAreaElement> = (e) => {
         this.setState({
             triggerWords: e.target.value,
         });
     };
 
-    updateTriggerWhen = (e) => {
+    updateTriggerWhen: ChangeEventHandler<HTMLSelectElement> = (e) => {
         this.setState({
-            triggerWhen: e.target.value,
+            triggerWhen: parseInt(e.target.value, 10),
```

Comment:
**Note:** I had assumed passing a `number` as `value` in the `option` element would allow us to just pass things through on the event. I was unable to get this working properly so opted to parse things out here rather then setting a string and parsing in `handleSubmit` above. Let me know if there's a different way to handle this.

Reply:
this is correct because the option value are supposed to be string and in the value of the select we need to typecast it back to string

Reply:
Sorry a little confused.

Would we prefer to update `triggerWhen` in `State` to `string` and typecast from `number` to `string` in `getStateFromHook()` and then from `string` to `number` in `handleSubmit`?

Or is it ok keeping the state value a `number` and then typecasting from `string` to `number` here?

Reply:
@M-ZubairAhmed wanted to ping again to see if there are additional changes needed from me before merging, when you get a chance. Thank you!

Reply:
@austin-denoble thank you this looks and works correctly so approving.